frontend/src/metabase/ui/components/utils/DelayGroup/DelayGroup.tsx

```diff
@@ -0,0 +1,61 @@
+import {
+  createContext,
+  useMemo,
+  useState,
+  useRef,
+  useEffect,
+  useContext,
+} from "react";
+
+type DelayGroupContext = {
+  shouldDelay: boolean;
+  onOpen: () => void;
+  onClose: () => void;
+};
+
+const context = createContext<DelayGroupContext>({
+  shouldDelay: true,
+  onOpen: () => undefined,
+  onClose: () => undefined,
+});
+
+type DelayGroupProps = {
+  timeout?: number;
+  children: React.ReactNode;
+};
+
+const DEFAULT_TIMEOUT = 500;
+
+export function DelayGroup({
+  children,
+  timeout = DEFAULT_TIMEOUT,
+}: DelayGroupProps) {
+  const [shouldDelay, setShouldDelay] = useState(true);
+
+  const t = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
```

Comment:
nit: :pray: 

_I missed that one in original review_

```suggestion
  const timeoutIdRef = useRef<ReturnType<typeof setTimeout> | undefined>(undefined);
```